src/kudu/consensus/consensus_queue.cc

```diff
@@ -999,8 +999,62 @@ void PeerMessageQueue::AdvanceMajorityReplicatedWatermarkFlexiRaft(
 }
 
 std::map<std::string, int> PeerMessageQueue::GetDataCommitQuorum() {
+  // Map to store the number of voters in each region from the active config.
+  std::map<std::string, int> voter_distribution;
+
+  // Return value denoting the number of votes required in each region.
   std::map<std::string, int> data_commit_quorum;
-  // TODO(ritwikyadav) : Populate quorum from cmeta_
+
+  CHECK(queue_state_.active_config->has_commit_rule());
+
+  // Set to help process each peer only once.
+  std::set<std::string> unique_uuids;
+
+  // Step 1: Compute total number of voters in each region.
+  for (const RaftPeerPB& peer : queue_state_.active_config->peers()) {
+    // Only consider peers that are voters.
+    if (!peer.has_member_type() ||
+        peer.member_type() != RaftPeerPB::VOTER) {
+      continue;
+    }
+    CHECK(peer.has_permanent_uuid() && peer.has_attrs() &&
```

Comment:
nit: this will crash the server. I am not sure every peer having a region is strictly enforced?